pyfat/PyFAT.py

```diff
@@ -0,0 +1,143 @@
+
+    #################################################
+    #   Python FATigue Test Data Analysis Package   #
+    #                  Version 1.0                  #
+    #                                               #
+    #                  Kevan Gahan                  #
+    #################################################
+
+#-----------------------#
+from pathlib import Path
+import os
+import argparse
+#--------#
+import monotonic
+import get_channels
+#-----------------------#
+
+
+def io_sorter(input_file):
+    """Reads the input file for the data input location and the 
+    output location (where to save results to)."""
+
+    with open(Path(input_file),'r') as input_file:
+        lines = input_file.readlines()
+
+    for line in lines:
+        #Remove and spaces...
+        std_line = line.replace(" ","")
+        #Assign file paths...
+        if std_line.startswith("INPUT="):
+            input_loc = std_line[6:]
+        elif std_line.startswith("OUTPUT="):
+            output_loc = std_line[7:]
+        elif std_line.startswith("#") or std_line.startswith(""):
+            pass
+        else:
+            raise AttributeError(
+                "Input file is not formatted correctly. Refer to Documentation."
+            )
+
+    return input_loc, output_loc
+
+
+def analysis_iteration(input_path, output_path, monotonic_bool, fatigue_bool):
+    """performs iteration of analysis. Takes in user input for analysis 
+    type and directory path to data, performs analysis iteration for 
+    selected analysis type. Saves results to user-defined save location."""
+
+    #Get list of data files in input directory
+    input_dir = str(input_path).strip()
+    files = next(os.walk(input_dir))[2]
+    
+    #Remove hidden folders/files from list
+    for filename in files:
+        if filename.startswith("."):
+            files.remove(filename)
+    
+    #Get channel names from first file 
```

Comment:
Can you only do either monotonic or fatigue but not both? If so you could just have one input variable (i.e. analysis_type which could be either monotonic or fatigue) and then just check which type it is and then raise an error if its not one of the acceptable types.